edu.cuny.hunter.streamrefactoring.core/src/edu/cuny/hunter/streamrefactoring/core/analysis/Util.java

```diff
@@ -407,13 +410,31 @@ static int getLineNumberFromIR(IBytecodeMethod method, SSAInstruction instructio
 		return lineNumberFromIR;
 	}
 
+	static private LinkedList<Value> checkInstructions = new LinkedList<>();
+
+	static Collection<TypeAbstraction> getPossibleStreamTypes(int valueNumber, TypeInference inference) {
+		checkInstructions.clear();
+		return getPossibleTypes(valueNumber, inference);
+	}
+
 	static Collection<TypeAbstraction> getPossibleTypes(int valueNumber, TypeInference inference) {
 		Set<TypeAbstraction> ret = new HashSet<>();
-		Value value = inference.getIR().getSymbolTable().getValue(valueNumber);
+		Value value;
+		try {
+			value = inference.getIR().getSymbolTable().getValue(valueNumber);
+		} catch (IllegalArgumentException exception) {
+			return ret;
+		}
 
 		// TODO: Should really be using a pointer analysis here rather than
 		// re-implementing one using PhiValue.
 		if (value instanceof PhiValue) {
+
+			if (checkInstructions.contains(value))
```

Comment:
I'm not convinced that this is needed. Also, there's no comments as to why it is needed.

Reply:
If there is no this, the method could be recursively called again and again and will never return.

Reply:
Are you experiencing infinite recursion with this method?

Reply:
OK I tried removing it and it looks like it is happening: https://travis-ci.com/ponder-lab/Optimize-Java-8-Streams-Refactoring/builds/82575224?utm_source=email&utm_medium=notification

Reply:
So, it's not that there "could be," *there is*.